light_curves/scale_up.md

```diff
@@ -0,0 +1,558 @@
+$ bash code_src/helpers/scale_up.sh \
+    -r "$run_id" \
+    -j '{"get_samples": {"SDSS": {"num": 500000}}, "archives": {"ZTF": {"nworkers": 8}}}' \
+    -a "core"  # shortcut for "Gaia HEASARC IceCube WISE ZTF"
+```
+
++++
+
+The script will run the 'get sample' job, then launch the archive query jobs in parallel and exit.
+Archive jobs will continue running in the background until they either complete or encounter an error.
+Example 2 shows how to load the data.
+
+Command output from 2024/03/01 logs:
+
+```{code-cell}
+!cat output/lightcurves-demo-SDSS-500k/logs/scale_up.sh.log
+```
+
+### Cancel
+
++++
+
+You can cancel jobs at any time.
+
+If the script is still running, press `Control-C`.
+
+If the script has exited, there are two options.
+
+1) To cancel an individual job, kill the job's process using:
+
++++
+
+```bash
+$ pid=0000  # get the number from script output
+$ kill $pid
+```
+
++++
+
+2) To cancel the entire run, use the `-k` (kill) flag:
+
++++
+
+```bash
+$ bash code_src/helpers/scale_up.sh -r "$run_id" -k
+```
+
++++
+
+### Restart
+
++++
+
+If you want to restart and skip step(s) that previously completed, run the first command again and add one or both "overwrite" flags set to false:
+
++++
+
+```bash
+# use the same run_id as before
+$ bash code_src/helpers/scale_up.sh \
+    -r "$run_id" \
+    -j '{"get_samples": {"SDSS": {"num": 500000}}, "archives": {"ZTF": {"nworkers": 8}}}' \
+    -a "core" \
+    -d "overwrite_existing_sample=false" \
+    -d "overwrite_existing_lightcurves=false"
+```
+
++++
+
+### Monitor
+
++++
+
+There are at least three places to look for information about a run's status.
+- Check the logs for job status or errors. The bash script will redirect stdout and stderr to log files and print out the paths for you.
+- Check for light curve (parquet) data. The script will print out the "parquet_dir". `ls` this directory. You will see a subdirectory for each archive call that has completed successfully, assuming it retrieved data for the sample.
+-  Watch `top`. The script will print the job PIDs. The script can also monitor `top` for you and save the output to a log file.
+
++++
+
+#### Logs
+
+Gaia log from 2024/03/01 (success):
+
+```{code-cell}
+!cat output/lightcurves-demo-SDSS-500k/logs/gaia.log
+```
+
+ZTF log from 2024/03/01 (failure):
+
+```{code-cell}
+!cat output/lightcurves-demo-SDSS-500k/logs/ztf.log
+```
+
+The logs above show that the ZTF job loaded the light curve data successfully ("100%") but exited without writing the parquet file (no "Light curves saved" message like Gaia).
+The data was lost.
+There is also no indication of an error; the job just ends.
+We can spot what happened by looking at the `top` output.
+
++++
+
+#### `top`
+
+During a run, you can use the `-t` (top) flag to have the script monitor `top` for you and save the output to a log file.
+To capture the complete run, open a second terminal and call the script with `-t` right after launching the run.
```

Comment:
I don't understand this.  If i just run the bash scale_up.sh with a -t will it do to the top monitoring, or do I need to call the bash script twice in two different terminal windows?

Reply:
To have the bash script monitor top, it needs to be called separately from the call that launches the run. I have clarified the text in various places in the notebook.